### ISSUE TITLE
Nx Console v18.18.0 cannot find any projects in my workspace

### ISSUE THREAD

**Issue by jorgeruvalcaba** [nrwl/nx-console#2081]
## Current Behavior

![image](https://github.com/nrwl/nx-console/assets/7416811/d464a245-4615-41b3-a2e3-37512a39a3ca)

## Expected Behavior

Nx Console can find projects in my Nx workspace.

## Steps to Reproduce

This is happening on a fresh install of Nx Console

### Failure Logs / Images / Videos

In the Nx Console Client logs:

```
[Nxls] - 2024-04-03T17:17:42.483Z - Retrieving workspace configuration

[Nxls] - 2024-04-03T17:17:42.486Z - Unable to load the undefined dependency from the workspace. Please ensure that the proper dependencies are installed locally.

[Nxls] - 2024-04-03T17:17:42.486Z - Unable to get nx workspace configuration: local Nx dependency not found
```

Local dependencies:

```
├── @angular/cli@17.1.2
├── @nrwl/cli@15.9.3
├── npm@8.9.0
└── typescript@5.3.3
```

### Environment

```
 Node : 20.11.0
   OS   : win32 x64
   npm  : 8.9.0

   nx : 13.8.2
   @nrwl/angular : 13.8.2
   @nrwl/cli : 13.8.2
   @nrwl/cypress : 13.8.2
   @nrwl/devkit : 13.8.2
   @nrwl/eslint-plugin-nx : 13.8.2
   @nrwl/jest : 13.8.2
   @nrwl/linter : 13.8.2
   @nrwl/storybook : 13.8.2
   @nrwl/tao : 13.8.2
   @nrwl/web : undefined
   @nrwl/workspace : 13.8.2
   typescript : 4.4.4
   rxjs : 7.5.7
```

- Nx Console version: v18.18.0
- VSCode version:

```
Version: 1.87.2 (user setup)
Commit: 863d2581ecda6849923a2118d93a088b0745d9d6
Date: 2024-03-08T15:20:17.278Z
Electron: 27.3.2
ElectronBuildId: 26836302
Chromium: 118.0.5993.159
Node.js: 18.17.1
V8: 11.8.172.18-electron.0
OS: Windows_NT x64 10
```

**Comment by jorgeruvalcaba** [comment]
Had to do a downgrade to v18.14.1. `nx-console` doesn't work in any version after it for my project.

**Comment by MaxKless** [comment]
I'm on it, will have a fix PR soon :)

**Comment by serjmac** [comment]
> Had to do a downgrade to v18.14.1. `nx-console` doesn't work in any version after it for my project.

Same issue here. I work on a 13.x workspace which fails, while in a 12.x workspace doesn't fail.

For PHPStorm, the latest plugin which doesn't has this issue is v1.17

**Comment by MaxKless** [comment]
This will be fixed with https://github.com/nrwl/nx-console/pull/2084

I also added some e2e tests so it doesn't break anymore in the future. Nx 12 is not supported by Nx Console anymore.

**Comment by jorgeruvalcaba** [comment]
> This will be fixed with #2084
> 
> I also added some e2e tests so it doesn't break anymore in the future. Nx 12 is not supported by Nx Console anymore.

Thanks for checking it out! Appreciate it!

**Comment by DannyvanHolten** [comment]
I'm having the same problem for IntelliJ V1.16.1, could it be that it's not fixed for IntelliJ yet?

**Comment by DannyvanHolten** [comment]
I am sorry. I did not do my own proper research. Restarting my IDE worked. So I think Nx console requires an initialization after installation.

**Comment by jackofsometrade** [comment]
I'm encountering this issue today. I've restarted my IDE multiple times, deleted all caches, removed all changes to my local code, still  "couldn't find any projects in this workspace". I'm getting really tired of stuff like this in `nx`.

**Comment by MaxKless** [comment]
are there any errors shown in that screen? Could you provide some logs? 
- Go to `Output -> Nx Language Server` and paste what you see there. 
- Run `Toggle Developer Tools` and paste what you see there too

Thank you!

**Comment by josue22almo** [comment]
I am facing the same issue at VS Code (VSCode Version: 1.93.1)

This is the output of Nx Console

![Image](https://github.com/user-attachments/assets/01b13f14-16f8-4fcd-b4de-a232b277cebe)


Dependencies:

```
"@nx/cypress": "20.2.2",
    "@nx/detox": "20.2.2",
    "@nx/esbuild": "20.2.2",
    "@nx/eslint-plugin": "20.2.2",
    "@nx/expo": "20.2.2",
    "@nx/jest": "20.2.2",
    "@nx/js": "20.2.2",
    "@nx/linter": "19.8.4",
    "@nx/nest": "20.2.2",
    "@nx/node": "20.2.2",
    "@nx/react": "20.2.2",
    "@nx/react-native": "20.2.2",
    "@nx/storybook": "20.2.2",
    "@nx/vite": "20.2.2",
    "@nx/web": "20.2.2",
    "@nx/webpack": "20.2.2",
    "@nx/workspace": "20.2.2",
    "nx": "20.2.2",
    "nx-cloud": "latest",
```


The project is listed without any app or lib.

![Image](https://github.com/user-attachments/assets/21a75a50-28c3-4d1b-8dc9-ca3ad10c07a5)

**Comment by MaxKless** [comment]
please include the output of the Nx Language Server tab and your nx report. You can also look at the Errors tab to see any project graph errors from nx

**Comment by josue22almo** [comment]
### Nx report

```
 NX   Report complete - copy this into the issue template

Node           : 22.10.0
OS             : darwin-arm64
Native Target  : aarch64-macos
yarn           : 1.22.22

nx (global)            : 20.0.0
nx                     : 20.2.2
@nx/js                 : 20.2.2
@nx/jest               : 20.2.2
@nx/linter             : 19.8.4
@nx/eslint             : 20.2.2
@nx/workspace          : 20.2.2
@nx/cypress            : 20.2.2
@nx/detox              : 20.2.2
@nx/devkit             : 20.2.2
@nx/esbuild            : 20.2.2
@nx/eslint-plugin      : 20.2.2
@nx/expo               : 20.2.2
@nx/module-federation  : 20.2.2
@nx/nest               : 20.2.2
@nx/node               : 20.2.2
@nx/react              : 20.2.2
@nx/react-native       : 20.2.2
@nx/storybook          : 20.2.2
@nx/vite               : 20.2.2
@nx/web                : 20.2.2
@nx/webpack            : 20.2.2
nx-cloud               : 19.1.0
typescript             : 5.7.2
---------------------------------------
Registered Plugins:
@nx/expo/plugin
@nx/eslint/plugin
@nx/storybook/plugin
---------------------------------------
The following packages should match the installed version of nx
  - @nx/linter@19.8.4

To fix this, run `nx migrate nx@20.2.2`
```

### Nx Language Server

```
[Nxls] - 2024-12-18T08:48:53.465Z - stopping daemon with `nx daemon --stop`

[Nxls] - 2024-12-18T08:48:53.466Z - Using local Nx package at /Users/Documents/mi-gestor-app/node_modules/nx/src/utils/package-manager

[Nxls] - 2024-12-18T08:48:54.659Z - Unregistering file watcher

[Nxls] - 2024-12-18T08:48:54.835Z - Initializing Nx Language Server

[Nxls] - 2024-12-18T08:48:54.841Z - Retrieving workspace configuration for nx 20.2.2

[Nxls] - 2024-12-18T08:48:54.847Z - Using local Nx package at /Users/Documents/mi-gestor-app/node_modules/nx/src/utils/output.js

[Nxls] - 2024-12-18T08:48:54.911Z - Using local Nx package at /Users/Documents/mi-gestor-app/node_modules/nx/src/project-graph/file-map-utils.js

[Nxls] - 2024-12-18T08:48:54.911Z - Using local Nx package at /Users/Documents/mi-gestor-app/node_modules/nx/src/project-graph/project-graph.js

[Nxls] - 2024-12-18T08:48:54.911Z - Using local Nx package at /Users/Documents/mi-gestor-app/node_modules/nx/src/daemon/client/client.js

[Nxls] - 2024-12-18T08:48:54.916Z - Using local Nx package at /Users/Documents/mi-gestor-app/node_modules/nx/src/utils/package-manager

[Nxls] - 2024-12-18T08:48:55.672Z - createProjectGraphAndSourceMapsAsync

[Nxls] - 2024-12-18T08:48:57.382Z - createProjectGraphAndSourceMapsAsync successful

[Nxls] - 2024-12-18T08:48:57.403Z - Resetting daemon client

[Nxls] - 2024-12-18T08:48:57.403Z - Retrieved workspace configuration in: 2562.2619999999997 ms

[Nxls] - 2024-12-18T08:48:57.957Z - Using local Nx package at /Users/Documents/mi-gestor-app/node_modules/nx/src/daemon/client/client.js

[Nxls] - 2024-12-18T08:48:57.957Z - Initializing daemon watcher 

[Nxls] - 2024-12-18T08:48:58.047Z - Initialized daemon watcher

[Nxls] - 2024-12-18T08:48:58.065Z - Retrieving workspace configuration for nx 20.2.2

[Nxls] - 2024-12-18T08:48:58.068Z - Using local Nx package at /Users/Documents/mi-gestor-app/node_modules/nx/src/utils/nx-cloud-utils

[Nxls] - 2024-12-18T08:48:58.148Z - Using local Nx package at /Users/Documents/mi-gestor-app/node_modules/nx/src/project-graph/project-graph.js

[Nxls] - 2024-12-18T08:48:58.148Z - Using local Nx package at /Users/Documents/mi-gestor-app/node_modules/nx/src/utils/output.js

[Nxls] - 2024-12-18T08:48:58.148Z - Using local Nx package at /Users/Documents/mi-gestor-app/node_modules/nx/src/project-graph/file-map-utils.js

[Nxls] - 2024-12-18T08:48:58.148Z - Using local Nx package at /Users/Documents/mi-gestor-app/node_modules/nx/src/daemon/client/client.js

[Nxls] - 2024-12-18T08:48:58.151Z - createProjectGraphAndSourceMapsAsync

[Nxls] - 2024-12-18T08:48:58.161Z - Making recent CIPE request

[Nxls] - 2024-12-18T08:48:58.237Z - createProjectGraphAndSourceMapsAsync successful

[Nxls] - 2024-12-18T08:48:58.247Z - Resetting daemon client

[Nxls] - 2024-12-18T08:48:58.247Z - Retrieved workspace configuration in: 182.04783300000008 ms

[Nxls] - 2024-12-18T08:48:58.560Z - Unregistering file watcher

[Nxls] - 2024-12-18T08:48:58.564Z - Using local Nx package at /Users/Documents/mi-gestor-app/node_modules/nx/src/daemon/client/client.js

[Nxls] - 2024-12-18T08:48:58.564Z - Initializing daemon watcher 

[Nxls] - 2024-12-18T08:48:58.620Z - Initialized daemon watcher

[Nxls] - 2024-12-18T08:48:58.645Z - Using local Nx package at /Users/Documents/mi-gestor-app/node_modules/nx/src/utils/nx-cloud-utils

[Nxls] - 2024-12-18T08:48:58.650Z - Making claimed request

[Nxls] - 2024-12-18T08:48:58.834Z - Error from https://cloud.nx.app/nx-cloud/is-workspace-claimed: 

```

Running Nx graph from the CLI works properly.

Running VS Code `Nx: show full project graph` doesn't/
